common/src/main/java/com/revenuecat/purchases/common/BC5StoreProduct.kt

```diff
@@ -0,0 +1,41 @@
+package com.revenuecat.purchases.common
+
+import android.os.Parcelable
+import com.android.billingclient.api.ProductDetails
+import com.revenuecat.purchases.ProductType
+import com.revenuecat.purchases.models.StoreProduct
+import com.revenuecat.purchases.parceler.JSONObjectParceler
+import kotlinx.parcelize.Parcelize
+import kotlinx.parcelize.RawValue
+import kotlinx.parcelize.TypeParceler
+import org.json.JSONObject
+
+@Parcelize
+@TypeParceler<JSONObject, JSONObjectParceler>()
+data class BC5StoreProduct(
+    override val sku: String,
+    override val type: ProductType,
+    override val price: String,
+    override val priceAmountMicros: Long,
+    override val priceCurrencyCode: String,
+    override val originalPrice: String?,
+    override val originalPriceAmountMicros: Long,
+    override val title: String,
+    override val description: String,
+    override val subscriptionPeriod: String?,
+    override val freeTrialPeriod: String?,
+    override val introductoryPrice: String?,
+    override val introductoryPriceAmountMicros: Long,
+    override val introductoryPricePeriod: String?,
+    override val introductoryPriceCycles: Int,
+    override val iconUrl: String,
+    override val originalJson: JSONObject,
+    val productDetails: @RawValue ProductDetails?,
+    val offerToken: String?,
+    val pricingPhases: @RawValue ProductDetails.PricingPhases
```

Comment:
which pricing phases are we thinking we'd parse into here? the base plan should only have on pricing phase, and the offers would have multiple

Reply:
That's a good point - my original idea was to duplicate the storeProduct, so you could have multiple storeproducts mapping to the same package, one per each offer. (each one will have its own offerToken which is unique to storeproduct)
Not sure what extra information we'll need to attach to the storeproduct so that for developer eligibility it actually makes sense.
So I think what we'll need in the mapping phase is that we could potentially duplicate a package (not sure if any of our model might break there), or map a package to multiple storeproducts.